qiskit/providers/backend.py

```diff
@@ -229,3 +233,239 @@ class can handle either situation.
+        conditions."""
+        pass
+
+    @property
+    @abstractmethod
+    def max_shots(self) -> int:
+        """Return the maximum number of shots supported by the backend.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def max_circuits(self):
+        """The maximum number of circuits (or Pulse schedules) that can be
+        run in a single job.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @classmethod
+    @abstractmethod
+    def _default_options(cls):
+        """Return the default options
+
+        This method will return a :class:`qiskit.providers.Options`
+        subclass object that will be used for the default options. These
+        should be the default parameters to use for the options of the
+        backend.
+
+        Returns:
+            qiskit.providers.Options: A options object with
+                default values set
+        """
+        pass
+
+    def t1(self, qubit: Union[int, List[int]]) -> Union[float, List[float]]:
+        """Return the T1 time of a given qubit
+
+        Args:
+            qubit: The qubit index to get the T1 time for. If
+                a list is specified the output will be a list with the
+                T1 time for the specified qubits in the same order.
+
+        Returns:
+            t1: the T1 time for the specified qubit(s) in seconds.
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                t1 time for a qubit
+        """
+        raise NotImplementedError
+
+    def t2(self, qubit: Union[int, List[int]]) -> Union[float, List[float]]:
+        """Return the T2 time of a given qubit
+
+        Args:
+            qubit: The qubit index or indices to get the T2 time for. If
+                a list is specified the output will be a list with the
+                T2 time for the specified qubits in the same order.
+
+        Returns:
+            t2: the T2 time for the specified qubit(s) in seconds
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                t2 time for a qubit
+        """
+        raise NotImplementedError
+
+    def qubit_frequency(self, qubit: Union[int, List[int]]) -> Union[float, List[float]]:
+        """Return the frequency of a given qubit
+
+        Args:
+            qubit: The qubit index to get the frequency for. If
+                a list is specified the output will be a list with the
+                frequency for the specified qubits in the same order.
+
+        Returns:
+            frequency: the frequency of the specified qubit(s) in Hz
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                qubit frequency
+        """
+        raise NotImplementedError
+
+    @property
+    def dt(self) -> float:
```

Comment:
`dt` and `dtm` are not necessary if the provider doesn't support pulse API. I guess they are added here to transpile circuit `Delay` instruction, but maybe the conversion of SI to cycle time can be offloaded to the provider.

Reply:
Yeah, I left this in here because I thought it was a necessary component for pulse support currently. It is optional (because pulse support is) so it raises `NotImplementedError` unless a particular backend overloads it (as part of indicating they have pulse support). I would like to keep everything in SI units it seems easier to me for users to reason about things in terms of seconds instead of dt and then the particular provider/backend will do the conversion as part of the job submission.

But that's probably a larger change than we can do here. I'm hoping to keep this PR to just defining the interface for backends and we can start working on updating things in terra to use these new interfaces over time (so having backwards compat with the current terra internals is needed).